src/cmds/sys/date.c

```diff
@@ -79,42 +79,52 @@ static void set_date(char *new_date) {
 	settimeofday(&tv, NULL);
 }
 
-static char *get_next_key(char *string) {
-	while(*string++) {
-		if('%' == *string) {
-			return ++string;
-		}
-	}
-	return NULL;
-}
-#define MAX_NANOSECONDS 999999999
-static int check_format(char *string) {
-	char *str;
-	if(string[0] != '+') {
-		printf("invalid format string use '+%%N'\n");
+static int show_fdate(char *fmt) {
+
+	if (fmt[0] != '+') {
+		printf("invalid format string, use '+FORMAT'\n");
 		return -1;
 	}
-	if(NULL != (str = get_next_key(string))) {
-		switch(str[0]) {
-		case 'N': {
-			struct timeval tv;
-			time64_t ns;
-
-			ktime_get_timeval(&tv);
-			ns = timeval_to_ns(&tv);
-			printf("%lld\n", ns % MAX_NANOSECONDS);
+
+	struct timeval tv;
+	gettimeofday(&tv, NULL);
+
+	char *c = fmt + 1;
+	while (*c) {
+		if (*c != '%') {
+			printf("%c", *c);
+			++c;
+			continue;
+		}
+
+		++c;
+		if (!c) {
```

Comment:
Should it be `!*c`?

Reply:
It should. That's the real value of code-review. Thank you!